### DISCUSSION TITLE
How to create a version 12 app?

### DISCUSSION THREAD

**Discussion by littlesandra88** [vercel/next.js#43454] [Help] [answered]
Dear all =)

Now that version 13 have been released I can't figure out how to create a new version 12 TypeScript app.

Does anyone know how to do that?

Hugs,
Sandra =)

**Comment by icyJoseph** [comment, marked as answer]
`npx create-next-app@latest` and it'll ask you if you want to use TS or not, it'll also ask if you want Eslint

For example using `yarn`:

```shell
yarn create next-app test-delete-me
yarn create v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...

success Installed "create-next-app@13.0.5" with binaries:
      - create-next-app
✔ Would you like to use TypeScript with this project? … No / Yes
✔ Would you like to use ESLint with this project? … No / Yes
Creating a new Next.js app in /my/local/path/test-delete-me.
```

**Reply by littlesandra88** [reply]
Yes, but it creates a next 13 app. Here is what ends up in `packages.json`

```
  "dependencies": {
    "next": "13.0.5",
    "react": "18.2.0",
    "react-dom": "18.2.0"
  },
```

Even if I do `npx create-next-app@^12 --ts` I get a next 13 app...

**Reply by icyJoseph** [reply]
Oh I see... you can run `npm install next@12` after you `cd` into the project.

Also downgrade `eslint-config-next` to `@12` and last but not least, remove `.next`, `node_modules` and reinstall everything again.

**Reply by icyJoseph** [reply]
Why do you need 12 in particular? `next/link` or the browserlist changes?

**Reply by icyJoseph** [reply]
@littlesandra88 the new changes need to be activated with an experimental flag, so it is not something to worry about. 

As far as next@13 pages is concerned there's no non-stable changes. There's however a slightly breaking change regarding `next/link`. The changes you've heard about are on something called `app` directory, which doesn't impact `pages`.

It should be enough with the changes I listed above. `npm install next@12` and `npm install --save-dev eslint-config-next@12`.

Let us know if it works, and don't forget to mark as answered if it does help out!

**Reply by littlesandra88** [reply]
Interesting. Does that mean, that all the current Next 12 videos on Youtube are still valid as long as I don't use the experimental flag?

**Reply by icyJoseph** [reply]
For following tutorial and such I'd say you should be good to go, except for the `next/link` behaviour, but there were some changes to be mindful about:

https://nextjs.org/blog/next-13#breaking-changes

You could also point to an old template, like:

```shell
npx create-next-app --example https://github.com/vercel/next.js/tree/57426a2849b259618cebca428ba125c75c8d2ab5/examples/with-typescript with-typescript
```

But that one has something wrong with the @types/react family of packages, you need to upgrade them to @18... 


All in all, I'd say just take next@13 because for the `pages` directory, all tutorials still apply, with the caveat of `next/link`.

**Comment by littlesandra88** [comment]
I see.

From my understanding next 13 have (a lot) of non stable changes, and since I am new to Next, I would like to be able to get as much help as possible from YouTube videos =)
